### PR TITLE
Allow to dump hashes for the custom zone

### PR DESCRIPTION
**Pull request by fyrchik**
Also, set addresses instead of hex-hashes for custom contracts.

### CODE REVIEW

**Comment by alexvanin** [comment]
> Also, set addresses instead of hex-hashes for custom contracts.

So all contract related `TXT` records for both custom and neofs contracts store string LE encoded addresses, right?

**Comment by fyrchik** [comment]
@alexvanin no, for custom we store the address only, for neofs we store the hex. In another PR we will store 2 TXT records for neofs contracts, this could need some changes in the NNS contract.

**Review by alexvanin — APPROVED** [review]
Looks good, didn't test.

**Comment by fyrchik** [comment]
I have tested `init`, `deploy` and `dump-hashes`. Today I have added 2 commits with refactoring @realloc  could you test this branch for your use-case?

**Review by realloc — APPROVED** [review]
Works for simple TX

File: cmd/neofs-adm/internal/modules/morph/initialize_nns.go [hunk 1]
```diff
@@ -84,83 +82,89 @@ func (c *initializeContext) setNNS() error {
...
+		emit.Instruction(w.BinWriter, opcode.INITSSLOT, []byte{1})
+		wrapRegisterScriptWithPrice(w, nnsHash, script)
+		script = w.Bytes()
+	}
+
+	return c.sendCommitteeTx(script, -1, true)
 }
 
-func (c *initializeContext) emitUpdateNNSGroupScript(bw *io.BufBinWriter, nnsHash util.Uint160, pub *keys.PublicKey) (int64, error) {
+func (c *initializeContext) emitUpdateNNSGroupScript(bw *io.BufBinWriter, nnsHash util.Uint160, pub *keys.PublicKey) (bool, bool, error) {
```
Comment (carpawell) [inline cmd/neofs-adm/internal/modules/morph/initialize_nns.go:102]: two bool vals returned, document it?

**Review by carpawell — APPROVED** [review]

File: cmd/neofs-adm/internal/modules/morph/initialize_nns.go [hunk 2]
```diff
@@ -84,83 +82,89 @@ func (c *initializeContext) setNNS() error {
...
+	return false, isAvail, nil
 }
 
 func getAlphabetNNSDomain(i int) string {
 	return alphabetContract + strconv.FormatUint(uint64(i), 10) + ".neofs"
 }
 
-func (c *initializeContext) nnsRegisterDomainScript(nnsHash, expectedHash util.Uint160, domain string, setPrice bool) ([]byte, bool, error) {
+// wrapRegisterScriptWithPrice wraps a given script with `getPrice`/`setPrice` calls for NNS.
+// It is intended to be used for a single transaction, and not as a part of other scripts.
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply cmd/neofs-adm/internal/modules/morph/initialize_nns.go:139]: Fixed

File: cmd/neofs-adm/internal/modules/morph/initialize_nns.go (hunk 1 above)
Comment (fyrchik) [reply cmd/neofs-adm/internal/modules/morph/initialize_nns.go:105]: Fixed

**Review by carpawell — APPROVED** [review]

**Pull request merged**